plugins/modules/azure_rm_openshiftmanagedclusterversion_info.py

```diff
@@ -0,0 +1,135 @@
+    location: centralus
+  register: ocp_versions
+
+- name: Print available openshift versions
+  debug:
+    msg: "{{ ocp_versions }}"
+'''
+
+RETURN = '''
+versions:
+    description:
+        - openshift versions values
+    returned: always
+    type: list
+'''
+
+import json
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_ext import AzureRMModuleBaseExt
+from ansible_collections.azure.azcollection.plugins.module_utils.azure_rm_common_rest import GenericRestClient
+
+
+class Actions:
+    NoAction, Create, Update, Delete = range(4)
+
+
+class AzureRMOpenShiftManagedClustersVersionInfo(AzureRMModuleBaseExt):
+    def __init__(self):
+        self.module_arg_spec = dict(
+            location=dict(
+                type='str', required=True
+            )
+        )
+
+        self.location = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.url = None
+        self.status_code = [200]
+
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = '2023-11-22'
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        self.mgmt_client = None
+        super(AzureRMOpenShiftManagedClustersVersionInfo, self).__init__(self.module_arg_spec, supports_check_mode=True, supports_tags=False)
+
+    def exec_module(self, **kwargs):
+
+        for key in self.module_arg_spec:
+            setattr(self, key, kwargs[key])
+
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+        self.results = self.get_versions()
+        return self.results
+
+    def get_versions(self):
+        response = None
+        resp_results = {}
+        results = {}
+        # prepare url
+        self.url = ('/subscriptions' +
+                    '/{{ subscription_id }}' +
+                    '/providers' +
+                    '/Microsoft.RedHatOpenShift' +
+                    '/locations' +
+                    '/{{ location }}' +
+                    '/openshiftversions')
+        self.url = self.url.replace('{{ subscription_id }}', self.subscription_id)
+        self.url = self.url.replace('{{ location }}', self.location)
+        self.log("Fetch versions of openshift cluster.")
+        try:
+            response = self.mgmt_client.query(self.url,
+                                              'GET',
+                                              self.query_parameters,
+                                              self.header_parameters,
+                                              None,
+                                              self.status_code,
+                                              600,
+                                              30)
+            resp_results = json.loads(response.text)
```

Comment:
Get 'response.text()' method return value,
```suggestion
            resp_results = json.loads(response.text())
```

Reply:
Hi @Fred-sun 

Unfortunately, I can't set the `text` as a function because there is no such function.

```
ipdb> response.text
'{\n    "value": [\n        {\n            "properties": {\n                "version": "4.12.25"\n            }\n        },\n        {\n            "properties": {\n                "version": "4.13.23"\n            }\n        },\n        {\n            "properties": {\n                "version": "4.14.16"\n            }\n        }\n    ]\n}\n'

ipdb> response.text()
*** TypeError: 'str' object is not callable
```

Instead, I could change it to use the `json()` function.
But it means that I will need to use the `json()` function without the `json.loads()` function that includes the response.
```
ipdb> response.json()
{'value': [{'properties': {'version': '4.12.25'}}, {'properties': {'version': '4.13.23'}}, {'properties': {'version': '4.14.16'}}]}

ipdb> json.loads(response.json())
*** TypeError: the JSON object must be str, bytes or bytearray, not dict
```

Please, share what would you prefer to be used here.

Thank you.

Reply:
@MaxBab I tested it locally and returned the following value:
```e.g.
response.text => <bound method _RequestsTransportResponseBase.text of <RequestsTransportResponse: 200 OK, Content-Type: appli    cation/json>>

response.text() => {
    "value": [
        {
            "properties": {
                "version": "4.12.25"
            }
        },
        {
            "properties": {
                "version": "4.13.23"
            }
        },
        {
            "properties": {
                "version": "4.14.16"
            }
        }
    ]
}

Reply:
@MaxBab If two states are returned, can we classify them:
```e.g.
if instance(response.text, str)
    resp_results = json.loads(response.text)
else:
    resp_results = json.loads(response.text())

Reply:
@Fred-sun 

I'm not sure how you got the response with `response.text()`.
But I like the idea of classification.
Will apply it.

Thanks for the suggestion.

Reply:
Change added.